Allow running more tests on nrf54h20dk/nrf54h20/cpuppr

nordic-krch opened the pull request:
`cpuppr` is a coprocessor which requires `cpuapp` to start it. When sysbuild is used then `vpr_launcher` application can be used for `cpuapp` to run `cpuppr`. Adding `Kconfig.sysbuild` to main test folder and enabling `vpr_launcher` for `nrf54h20dk/nrf54h20/cpuppr` there.

Additionally, overwriting default for `CONFIG_ASSERT` to `n` for that board when `ZTEST` is enabled as with asserts enabled most tests does not fit.

Combining that with #73790 allows to run most of sw tests on that target.

nordic-krch committed:
tests: kernel: interrupt: nested_irq: support nrf54h20_cpuppr

Add dedicated interrupt lines for nrf54h20_cpuppr. Similar exception
was already added to nrf54l15_flpr: 8742e2476.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

masz-nordic commented on tests/Kconfig.sysbuild:
```diff
@@ -0,0 +1,7 @@
+# For tests on cpuppr on nrf54h20 we need vpr_launcher application on cpuapp
+# which kickstarts cpuppr.
+config VPR_LAUNCHER
+	default y if BOARD_NRF54H20DK_NRF54H20_CPUPPR
```
Could you add L15 FLPR as well?

nashif replied:
hmmm, this is not great. You are adding platform specific build system files in a common location. This does not make sense at all.

nordic-krch replied:
Removed that part.

tejlmand commented on boards/nordic/nrf54h20dk/Kconfig.defconfig:
```diff
@@ -14,3 +14,11 @@ config BT_CTLR
+
+if BOARD_NRF54H20DK_NRF54H20_CPUPPR
+
+# As PPR has limited memory most of tests does not fit with asserts enabled.
+config ASSERT
+	default n if ZTEST
```
as most tests are actually smaller in footprint than actual samples, then why not simple disable `ASSERT` per default, and remove the `ZTEST` dependency (and perhaps even move it to the SoC implementation).

Especially as it will be clearer to a reader that `ASSERT` is `default n` in all cases for this board (or SoC CPU cluster if  moved).

Current construct might mislead the reader to think that `ASSERT` will by `y` when not `ZTEST`, which is actually not the case, ref:
https://github.com/zephyrproject-rtos/zephyr/blob/4d0a8c1e8b8316ec5071a3fcc98cd292d3bc4e6d/subsys/debug/Kconfig#L231-L233

nordic-krch replied:
done

tejlmand commented on boards/nordic/nrf54l15pdk/Kconfig.defconfig:
```diff
@@ -10,3 +10,11 @@ config ROM_START_OFFSET
+if BOARD_NRF54L15PDK_NRF54L15_CPUFLPR || BOARD_NRF54L15PDK_NRF54L15_CPUFLPR_XIP
+
+# As FLPR has limited memory most of tests does not fit with asserts enabled.
+config ASSERT
+	default n if ZTEST
```
other comment regarding ASSERT also applies here.

nordic-krch replied:
done

nordic-krch commented:
@kl-cruz @masz-nordic can you re-approve? I've added targets with flpr that were not available at the time of creation of that PR.

nordic-krch committed:
soc: nordic: Disable asserts on ppr and flpr

Asserts are by default enabled for tests but flpr and ppr are
small cores (<64k) and many tests does not fit in memory with
asserts enabled.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>
